File: app/src/main/java/com/gmail/fattazzo/formula1world/fragments/current/constructors/ConstructorItemView.java
package com.gmail.fattazzo.formula1world.fragments.current.constructors;

import android.content.Context;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.gmail.fattazzo.formula1world.R;
import com.gmail.fattazzo.formula1world.domain.F1Constructor;
import com.gmail.fattazzo.formula1world.utils.CountryNationality;
import com.gmail.fattazzo.formula1world.utils.ImageUtils;
import com.gmail.fattazzo.formula1world.utils.Utils;

import org.androidannotations.annotations.Bean;
import org.androidannotations.annotations.EViewGroup;
import org.androidannotations.annotations.ViewById;

@EViewGroup(R.layout.constructors_item_list)
public class ConstructorItemView extends LinearLayout {

    @Bean
    Utils utils;

    @Bean
    ImageUtils imageUtils;

    @ViewById(R.id.constructor_item_name)
    TextView nameView;

    @ViewById(R.id.constructor_item_flag)
    ImageView flagImageView;

    public ConstructorItemView(Context context) {
        super(context);
    }

    public void bind(F1Constructor constructor) {
        nameView.setText(constructor.name);

        CountryNationality countryNationality = utils.getCountryNationality(constructor.nationality);
        if (countryNationality != null) {
            flagImageView.setImageBitmap(imageUtils.getFlagForCountryCode(countryNationality.getAlpha2Code()));
        }
    }
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/fragments/current/constructors/ConstructorItemView.kt
/*
 * Project: total-gp-world
 * File: ConstructorItemView.kt
 *
 * Created by fattazzo
 * Copyright © 2018 Gianluca Fattarsi. All rights reserved.
 *
 * MIT License
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.gmail.fattazzo.formula1world.fragments.current.constructors

import android.content.Context
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import com.gmail.fattazzo.formula1world.R
import com.gmail.fattazzo.formula1world.domain.F1Constructor
import com.gmail.fattazzo.formula1world.service.DataService
import com.gmail.fattazzo.formula1world.utils.ImageUtils
import com.gmail.fattazzo.formula1world.utils.Utils
import org.androidannotations.annotations.Bean
import org.androidannotations.annotations.EViewGroup
import org.androidannotations.annotations.ViewById

@EViewGroup(R.layout.constructors_item_list)
open class ConstructorItemView(context: Context) : LinearLayout(context) {

    @Bean
    lateinit internal var utils: Utils

    @Bean
    lateinit internal var dataService: DataService

    @Bean
    lateinit internal var imageUtils: ImageUtils

    @ViewById(R.id.constructor_item_name)
    lateinit internal var nameView: TextView

    @ViewById(R.id.constructor_item_flag)
    lateinit internal var flagImageView: ImageView

    @ViewById
    lateinit internal var constructor_color: ImageView

    fun bind(constructor: F1Constructor) {
        nameView!!.text = constructor.name

        val countryNationality = utils!!.getCountryNationality(constructor.nationality)
        if (countryNationality != null) {
            flagImageView!!.setImageBitmap(imageUtils!!.getFlagForCountryCode(countryNationality.alpha2Code!!))
        }

        val color = dataService!!.loadContructorColor(constructor)
        constructor_color!!.setBackgroundColor(color)
    }
}